src/Raven.Server/Documents/Sharding/Queries/AbstractShardedQueryProcessor.cs

```diff
@@ -159,7 +159,7 @@ public virtual ValueTask InitializeAsync()
         else
             queryTemplate = Query.ToJson(Context);
 
-        if (Query.Metadata.IsCollectionQuery && Query.Metadata.DeclaredFunctions is null or { Count: 0 })
+        if (Query.Metadata.IsCollectionQuery && Query.Metadata.DeclaredFunctions is null or { Count: 0 } && Query.Metadata.HasIncludeOrLoad == false)
```

Comment:
does it mean that queries such as in the test we will send now to all shards?

Reply:
That is _not_ something that we want.
IIRC, we have specific handling for includes, for example. That would fetch the data if needed.

Reply:
`load`, however, is not something that we can process

Reply:
In terms of the final result, there is no difference here between sending the query to all shards or to the known shards.

Reply:
You are talking about include? You have a 2 step process.
First send to the known shards (usually 1) and see if that  include is there.
If not, then you go to the other one.

Reply:
I'm talking in general. Sending to all nodes or sending to known shards should give us the same result.
Sending to known shards is an optimization, which I believe we want in this case as well.

Reply:
Yes, but anyway `load` is known problematic in sharding, isn't it?
I thought we didn't support loading from a different shard / bucket?

Reply:
correct. for `load` (both indexing & query) we simply take what we have on that shard

Reply:
Given that, is there an actual problem here? Except maybe give a better error?
Since that isn't _meant_ to be suported?

Reply:
The following query is supported - and the behaviour is documented (`load` in a projection will work only on the same shard).
This is saving us from doing an `include` which would send the entire document so I don't think that we should error here.
```
from 'Orders' as o
load o.Company as c
select {
     Handled: o.Employee,
     CompName: c.Name
}
```
The reason why we didn't get the expected result is because of an optimization when doing `where id() == 'orders/830-A' which doesn't do a query but fetches the document directly.

Reply:
The problem is that `ExractIdsFromQuery` needs to also handle `load / include` clauses